src/UI/Component/Dropzone/File/File.php

```diff
@@ -0,0 +1,130 @@
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Component\Dropzone\File
+ */
+interface File extends Component, Droppable {
+
+	/**
+	 * Get a dropzone like this where the files are uploaded to the given URL.
+	 *
+	 * @param string $url
+	 * @return $this
+	 */
+	public function withUploadUrl($url);
+
+	/**
+	 * Get the upload URL where the files are uploaded.
+	 *
+	 * @return string
+	 */
+	public function getUploadUrl();
+
+	/**
+	 * Get a dropzone like this only accepting the submitted file types for uploading, e.g.
+	 * ['jpg', 'png', 'gif'] to allow some image formats.
+	 *
+	 * @param array $types
+	 * @return $this
+	 */
+	public function withAllowedFileTypes(array $types);
+
+	/**
+	 * @return array
+	 */
+	public function getAllowedFileTypes();
+
+	/**
+	 * Get a dropzone like this, restricting the max number of files that can be uploaded.
+	 *
+	 * @param int $max
+	 * @return $this
+	 */
+	public function withMaxFiles($max);
+
+	/**
+	 * Get the max number of files that can be uploaded.
+	 *
+	 * @return int
+	 */
+	public function getMaxFiles();
+
+	/**
+	 * Get a dropzone like this, restricting the max file size of the files to the given limit (in bytes).
+	 *
+	 * @param int $limit Max size for any file uploaded in bytes
+	 * @return $this
+	 */
+	public function withFileSizeLimit($limit);
+
+	/**
+	 * Get the max file size limit in bytes.
+	 *
+	 * @return int
+	 */
+	public function getFileSizeLimit();
+
+	/**
+	 * Get a dropzone like this, allowing to set the filename for each file being uploaded.
+	 * The custom file name is sent as POST parameter along with the uploaded file.
+	 *
+	 * @param bool $state True to enable custom file names
+	 * @return $this
+	 */
+	public function withCustomFileNames($state);
+
+	/**
+	 * Check if the dropzone supports to enter custom file names for each file being uploaded.
+	 *
+	 * @return bool
+	 */
+	public function allowCustomFileNames();
+
+	/**
+	 * Get a dropzone like this, allowing to set a description for each file being uploaded.
+	 * The description is sent as POST parameter along with the uploaded file.
+	 *
+	 * @param bool $state True to enable file descriptions
+	 * @return $this
+	 */
+	public function withFileDescriptions($state);
+
+	/**
+	 * Check if the dropzone supports to enter file descriptions for each file being uploaded.
+	 *
+	 * @return bool
+	 */
+	public function allowFileDescriptions();
+
+	/**
+	 * Get a dropzone like this where each uploaded file is identified over a given identifier.
+	 * The identifier corresponds to the key used to identify the files server side,
+	 * e.g. $_FILES[identifier]
+	 *
+	 * Note: If you use multiple file dropzones on the same page, you MUST use identifier in
+	 * order to identify an uploaded file. The default identifier is 'files'.
+	 *
+	 * @param string $identifier
+	 * @return $this
+	 */
+	public function withIdentifier($identifier);
```

Comment:
Note that https://github.com/ILIAS-eLearning/ILIAS/pull/583 uses withParameterName() for something very similar. I somehow prefer the later.